src/main/java/ladder/domain/generator/DirectionGenerator.java

```diff
@@ -0,0 +1,35 @@
+package ladder.domain.generator;
+
+import ladder.domain.model.Direction;
+
+import java.util.function.Supplier;
+
+/**
+ * FIXME: implements Generator<Direction>
+ * 인터페이스 제네릭에다가 enum 을 넣고 싶은데,
+ * 문법상 허용되지 않는다. 방법이 없을까?
+ */
+public class DirectionGenerator {
```

Comment:
```suggestion
public class DirectionGenerator implements Generator<Direction> {
```

 본래는 `implements Generator<Direction>` 를 통해 Generator 인터페이스로 통일을 하고 싶었지만,
Direction 이 enum 이라, 제네릭 자리에 넣으니 컴파일 오류가 납니다.
enum 을 generic 으로 활용하려면 어떻게 해야 좋을 지 조언 부탁드립니다.

Reply:
앞선 리뷰에서 언급했지만 DirectionGenerator와 다른 Generator는 성격이 다르다고 생각합니다.
아예 DirectionStrategy 라는 형태로 만들어보는건 어떨까요?